cluster.go

```diff
@@ -485,11 +526,19 @@ func (c *Cluster) doInner(a Action, addr, key string, ask bool, attempts int) er
 
 	err = p.Do(thisA)
 	if err == nil {
+		c.setClusterDown(false)
 		return nil
 	}
 
-	// if the error was a MOVED or ASK we can potentially retry
 	msg := err.Error()
```

Comment:
as long as the logic in this area is getting changed, might as well add:

```
if !errors.As(err, new(resp2.Error)) {
    return err
}
msg := err.Error()
```

I can't imagine what kind of network errors would start with strings like `CLUSTERDOWN` or `MOVED` or `ASK`, but might as well be safe.

Reply:
Done